libqtile/backend/x11/window.py

```diff
@@ -1327,9 +1327,6 @@ def focus(self, warp: bool = True) -> None:
         self.qtile.core._root.set_property("_NET_ACTIVE_WINDOW", self.window.wid)
         self._ungrab_click()
 
-        if self.group:
```

Comment:
I don't really have a good mental picture right now, but it feels like they focus the window directly to avoid recursion with the group focus method. Can't we do something like this:
```python
        if self.group and self.group.current_window is not self:
            self.group.focus(self)
```

Reply:
Yes. I think my theory is wrong. I don't see anything subscribing to that hook.

Reply:
Ok - this is not the right fix. It's "working" because, after the focus changes, we get an `EnterNotify` event which then sorts out the group focus point (if `follow_mouse_focus` is set). I could use that code instead, after I've tried yours.